### ISSUE TITLE
Pool CancellationTokenSource for ActiveHealthCheckMonitor

### ISSUE THREAD

**Issue by Kahbazi** [microsoft/reverse-proxy#1354]
Is it possible to use `ActivityCancellationTokenSource.Rent` for `ActiveHealthCheckMonitor`?
https://github.com/microsoft/reverse-proxy/blob/1855217be702f3a6d721095b36b07f5458ee24a5/src/ReverseProxy/Health/ActiveHealthCheckMonitor.cs#L170

**Comment by MihaZupan** [comment]
Yes, it should be possible to plug in `ActivityCancellationTokenSource.Rent` with `CT.None` as the linked token, but the benefits would be in the noise range.

Another place that may be more impactful is in [`StreamCopyHttpContent`](https://github.com/microsoft/reverse-proxy/blob/11a6942f4d6f1b30478a18c86bd2b80abc0df1e0/src/ReverseProxy/Forwarder/StreamCopyHttpContent.cs#L146) for HTTP/2 and 3, but that would require a different pooled CTS.
